src/components/MultiHopTrade/components/MultiHopTradeConfirm/components/Hop.tsx

```diff
@@ -95,26 +105,52 @@ export const Hop = ({
       default:
         return null
     }
-  }, [chevronDownIcon, chevronUpIcon, estimatedExecutionTimeMs, isOpen, onToggleIsOpen, txStatus])
+  }, [
+    chevronDownIcon,
+    chevronUpIcon,
+    tradeQuoteStep.estimatedExecutionTimeMs,
+    isOpen,
+    onToggleIsOpen,
+    txStatus,
+  ])
 
-  const stepperSteps = useMemo(
-    () =>
-      steps.map(({ title, stepIndicator, description, content, key }, index) => (
-        <Step key={key}>
-          <StepIndicator>{stepIndicator}</StepIndicator>
+  const initialApprovalRequirements = useAppSelector(selectInitialApprovalRequirements)
+  const isApprovalInitiallyNeeded = initialApprovalRequirements?.[hopIndex]
 
-          <Box flexShrink='0'>
-            <StepTitle>{title}</StepTitle>
-            {description && <StepDescription>{description}</StepDescription>}
-            {index === activeStep && content}
-            {index < steps.length - 1 && <Spacer height={6} />}
-          </Box>
-          <StepSeparator />
-        </Step>
-      )),
-    [activeStep, steps],
+  const activeStep = useMemo(() => {
+    switch (hopExecutionState) {
+      case HopExecutionState.Pending:
+        return -Infinity
+      case HopExecutionState.AwaitingApprovalConfirmation:
+      case HopExecutionState.AwaitingApprovalExecution:
+        return hopIndex === 0 ? 1 : 0
```

Comment:
Should we `findIndexOf()`, or consume an enum, to keep things more declarative?

Reply:
the steps here are the "stepper" steps rather than any of the steps in the state machine (confusing as hell). The logic here needs to be ultra custom for this component